### ISSUE TITLE
Some optimizations about "_generateAllPossibleConfigurations" function

### ISSUE THREAD

**Issue by foreverkenan** [cemulate/polyomino-solver#9]
When Polyomino is centrosymmetric or left right symmetric, repeated calculations are performed, resulting in multiple identical configs
![image](https://github.com/cemulate/polyomino-solver/assets/41415508/c355b9fd-de89-4919-b5f3-62b0882348a6)

**Comment by foreverkenan** [comment]
origin
![image](https://github.com/cemulate/polyomino-solver/assets/41415508/47e856dd-cf00-442c-aecb-4fb92b08e068)
change
![image](https://github.com/cemulate/polyomino-solver/assets/41415508/e44295f8-6a7c-4d2f-b401-6563d69ae9b3)

-----

origin
![image](https://github.com/cemulate/polyomino-solver/assets/41415508/f8430e2d-26ea-4654-9ed5-c904bcaf16f2)
change
![image](https://github.com/cemulate/polyomino-solver/assets/41415508/074890e6-7578-4a98-a06d-e0f8071a8038)


only change "_generateAllPossibleConfigurations(piece)" function

```
* _generateAllPossibleConfigurations(piece) {

        let save = new Set();

        for (let rotation of [ 0, ...(this.allowRotation ? [1, 2, 3] : []) ])
        {
            let c = piece.rotate(rotation).normalize();

            for (let reflected of [ false, ...(this.allowReflection ? [ true ] : []) ])
            {
                if (reflected) c = c.reflect().normalize();

                if(!save.empty)
                {
                    let prevSize = save.size;

                    for(let p of c.coords)
                        save.add(p.stringify);

                    if(prevSize == save.size)
                        continue;
                }
                else
                {
                    for(let p of c.coords)
                        save.add(p.stringify);
                }

                for (let dx = 0; dx < this.width; dx ++)
                {
                    for (let dy = 0; dy < this.height; dy ++)
                    {
                        let config = c;

                        config = config.translate(dx, dy);

                        if (this._fits(config)) yield config;
                    }
                }
            }
        }
    }
```

**Comment by foreverkenan** [comment]
I implemented this optimization in C++, but I am not familiar with JavaScript. Some of the features of C++and JavaScript are not the same, and I am unable to implement them

**Comment by cemulate** [comment]
Thanks for the idea! It wasn't at all bad to implement. On the following test case, this optimization cuts down the SAT problem from 2768 variables and 889052 clauses (originally) to 1692 variables and 358086 clauses (the SAT methods still take forever to solve this problem, however).

![image](https://github.com/cemulate/polyomino-solver/assets/342497/9722d082-37cb-4239-bd95-0698435e9a87)

Fixed by 1c824d8
